mlflow/metrics/base.py

```diff
@@ -14,3 +14,35 @@ class MetricValue:
     scores: List[float] = None
     justifications: List[float] = None
     aggregate_results: Dict[str, float] = None
+
+
+@dataclass
+class EvaluationExample:
+    """
+    Stores the sample example during few shot learning during LLM evaluation
+    """
+
+    input: str
+    output: str
+    score: float
+    justification: str = None
+    variables: Dict[str, str] = None
```

Comment:
Will this be a public / user-facing class (I think it is)? If so, can we add detailed docstrings with examples for each field?